spec/integration/decoding_payloads_spec.rb

```diff
@@ -0,0 +1,42 @@
+class TwitterSubscriber < ActionSubscriber::Base
+  def tweet
+    $messages << {
+      :decoded => payload,
+      :raw => raw_payload,
+    }
+  end
+end
+
+::ActionSubscriber.config.add_decoder("foo/type" => lambda{ |payload| :foo })
+
+describe "Payload Decoding", :integration => true do
+  let(:connection) { subscriber.connection }
+  let(:subscriber) { TwitterSubscriber }
+  let(:json_string) { '{"foo": "bar"}' }
+
+  it "decodes json by default" do
+    ::ActionSubscriber.auto_subscribe!
+    channel = connection.create_channel
+    exchange = channel.topic("events")
+    exchange.publish(json_string, :routing_key => "twitter.tweet", :content_type => "application/json")
+    sleep 0.1
+
+    expect($messages).to eq Set.new([{
+      :decoded => JSON.parse(json_string),
+      :raw => json_string,
+    }])
+  end
+
+  it "can be given custom decoders" do
```

Comment:
I'd prefer to tie the adding of a custom decoder directly to this spec. When I looked through it initially (on my phone), it wasn't clear where the custom decoder was being defined, which also made it more difficult to decipher how the expected decoded value was being determined. Perhaps we can even unload/reset it after the specs run to ensure we are getting an isolated test case?